src/Voxel.MiddyNet.HttpJsonBodyParserMiddleware/HttpJsonBodyParserMiddleware.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Text;
+using Amazon.Lambda.APIGatewayEvents;
+using Newtonsoft.Json;
+using System.Threading.Tasks;
+
+namespace Voxel.MiddyNet.HttpJsonBodyParserMiddleware
+{
+    public class HttpJsonBodyParserMiddleware<T> : ILambdaMiddleware<APIGatewayProxyRequest, APIGatewayProxyResponse>
+    {
+        public Task Before(APIGatewayProxyRequest lambdaEvent, MiddyNetContext context)
+        {
+            if (!HasJsonContentHeaders(lambdaEvent))
+            {
+                context.AdditionalContext.Add("Body", lambdaEvent.Body);
+                return Task.CompletedTask;
+            }
+
+            if (lambdaEvent.IsBase64Encoded)
+            {
+                lambdaEvent.Body = Encoding.UTF8.GetString(Convert.FromBase64String(lambdaEvent.Body));
+            }
+           
+            T source;
+            try
+            {
+                source = JsonConvert.DeserializeObject<T>(lambdaEvent.Body);
+            }
+            catch (JsonReaderException)
+            {
+                throw new Exception("Content type defined as JSON but an invalid JSON was provided");
```

Comment:
Why are we throwing a generic exception. Wouldn´t be better to just let the original exception bubble up?

Reply:
Correct, remove the catch and let original exception bubble up.